### PR TITLE
Support installing from a local disk image

### PR DESCRIPTION
**Pull request by dm0-**
This is mostly to prepare for Packet to install pre-relase images that we need to get onto the systems by some other manner.

### CODE REVIEW

File: bin/coreos-install [hunk 1]
```diff
@@ -35,7 +35,7 @@ for f in /usr/share/oem/oem-release /etc/oem-release; do
     fi
 done
 
-USAGE="Usage: $0 [-V version] [-d /dev/device]
+USAGE="Usage: $0 [-C channel] [-d /dev/device]
```
Comment (bgilbert) [inline bin/coreos-install]: Good catch.

**Comment by bgilbert** [comment]
I typically use `bs=1M`.  It's mostly a question of system call overhead; I haven't checked in a long time, but the default 512 bytes used to significantly degrade throughput.

**Comment by dm0-** [comment]
Okay, it's set to `1M`.

File: bin/coreos-install (hunk 1 above)
Comment (bgilbert) [inline bin/coreos-install]: I just noticed: `-d` is required; we should take it out of brackets.

Comment (dm0-) [inline bin/coreos-install]: Done.

File: bin/coreos-install [hunk 2]
```diff
@@ -379,100 +377,132 @@ if [[ -n "${IGNITION}" ]]; then
     fi
 fi
 
-if [[ -n "${OEM_ID}" ]]; then
-    IMAGE_NAME="coreos_production_${OEM_ID}_image.bin.bz2"
-else
-    IMAGE_NAME="coreos_production_image.bin.bz2"
-fi
+function write_to_disk() {
+    trap 'error_output ; wipefs --all --backup "${DEVICE}" ; rm -rf "${WORKDIR}"' EXIT
```
Comment (bgilbert) [inline bin/coreos-install]: This doesn't wipefs if there's a bad GPG signature or fetch timeout.

**Comment by dm0-** [comment]
Alright, I've just made it `wipefs` on any failure.

**Comment by leroix** [comment]
@dm0- since this was merged I'm experiencing:

```
tom stdout Current version of CoreOS Container Linux stable is 1409.5.0
tom stdout Downloading the signature for https://stable.release.core-os.net/amd64-usr/1409.5.0/coreos_production_image.bin.bz2...
tom stdout Downloading, writing and verifying coreos_production_image.bin.bz2...
tom stdout /dev/sda: 8 bytes were erased at offset 0x00000200 (gpt): 45 46 49 20 50 41 52 54
tom stdout /dev/sda: 2 bytes were erased at offset 0x000001fe (PMBR): 55 aa
tom stdout /dev/sda: calling ioctl to re-read partition table: Success
tom stdout 
tom stderr 2017-06-30 03:36:45 URL:https://stable.release.core-os.net/amd64-usr/1409.5.0/coreos_production_image.bin.bz2.sig [543/543] -> "/tmp/coreos-install.svcpSyQ0ds/coreos_production_image.bin.bz2.sig" [1]
tom stderr 2017-06-30 03:38:11 URL:https://stable.release.core-os.net/amd64-usr/1409.5.0/coreos_production_image.bin.bz2 [289240133/289240133] -> "-" [1]
tom stderr gpg: Signature made Thu 22 Jun 2017 10:26:47 PM UTC using RSA key ID 1CB5FA26
tom stderr gpg: key 93D2DCB4 marked as ultimately trusted
tom stderr gpg: checking the trustdb
tom stderr gpg: 3 marginal(s) needed, 1 complete(s) needed, PGP trust model
tom stderr gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
tom stderr gpg: Good signature from "CoreOS Buildbot (Offical Builds) <buildbot@coreos.com>"
tom stderr mount: wrong fs type, bad option, bad superblock on ,
tom stderr        missing codepage or helper program, or other error
tom stderr 
tom stderr        In some cases useful info is found in syslog - try
tom stderr        dmesg | tail or so.
tom stderr Error: return code 32 from mount "${OEM_DEV}" "${WORKDIR}/oemfs"
tom stderr 
```
when running
```
> $HOME/coreos-install -d /dev/sda -i ignition.json
```

The mount error didn't happen before this merge. I'll continue investigating.

**Comment by bgilbert** [comment]
It appears `install_from_url` returns before `write_to_disk` finishes, so we start looking for the OEM partition before the partition table has been reread.

**Comment by bgilbert** [comment]
@leroix That should be fixed by #243.  Thanks for reporting!

**Comment by leroix** [comment]
Excellent! Thanks so much!

**Pull request merged**